File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for num in nums:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        sorted_dict = sorted(count.items() , key=lambda x : x[1] , reverse=True)
        
        answer = []
        for i in range(k):
            answer.append(sorted_dict[i][0])
        
        return answer
                
        

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
import java.util.*;
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer , Integer> map = new HashMap<>();
        PriorityQueue<Map.Entry<Integer , Integer>> pq = new PriorityQueue<>((a , b) -> a.getValue() - b.getValue());

        for(int num : nums){
            map.put(num , map.getOrDefault(num , 0) + 1);
        }
        
        for(Map.Entry<Integer , Integer> entry : map.entrySet()){
            pq.offer(entry);

            if(pq.size() > k){
                pq.poll();
            }
        }

        int[] answer = new int[k];
        for(int i = k - 1; i >= 0; i--){
            answer[i] = pq.poll().getKey();
        }

        return answer;
    }
}